src/celestia/scriptmenu.cpp

```diff
@@ -35,47 +44,60 @@ static void process(const fs::path& p, vector<ScriptMenuItem>* menuItems)
     // Scan the script file for metainformation. At the moment,
     // the only thing searched for is the script title, which must
     // appear on the first line after the string 'Title:'
-    ifstream in(p);
-    if (in.good())
-    {
-        ScriptMenuItem item;
-        item.filename = p;
+    std::ifstream in(p);
+    if (!in.good())
+        return;
 
-        // Read the first line, handling various newline conventions
-        char firstLineBuf[512];
-        size_t count = 0;
-        while (count < sizeof(firstLineBuf) - 1 && in.good())
-        {
-            int c = in.get();
-            if (c == '\n' || c == '\r')
-                break;
-            firstLineBuf[count++] = c;
-        }
+    ScriptMenuItem& item = menuItems.emplace_back();
+    item.title = p.filename().string();
+    item.filename = p;
 
-        string firstLine(firstLineBuf, count);
-        auto titlePos = firstLine.find(TitleTag);
+    // Read the first line, handling various newline conventions
+    std::array<char, 512> buffer;
+    in.read(buffer.data(), buffer.size());
+    std::string_view line;
+    if (!in.good() && !in.eof())
+        return;
 
-        // Skip spaces after the Title: tag
-        if (titlePos != string::npos)
-            titlePos = firstLine.find_first_not_of(' ', titlePos + (sizeof(TitleTag) - 1));
+    line = std::string_view(buffer.data(), static_cast<std::size_t>(in.gcount()));
 
-        if (titlePos != string::npos)
-        {
-            item.title = firstLine.substr(titlePos);
-        }
-        else
-        {
-            // No title tag--just use the filename
-            item.title = p.filename().string();
-        }
-        menuItems->push_back(item);
-    }
+    // Only consider the first line
```

Comment:
it seems that code tries to implement https://en.cppreference.com/w/cpp/io/basic_istream/getline